dev/ci/.gitlab-ci.yml

```diff
@@ -1,69 +1,64 @@
+# ==========================================================================
+# Main GitLab CI Configuration for the global-workflow Project
+# ==========================================================================
+#
+# This is the primary CI/CD configuration file that orchestrates the GitLab CI
+# testing framework for the global-workflow project. It defines pipeline stages
+# and includes specialized configuration files for different testing aspects.
+#
+# The pipeline architecture supports:
+#   1. Multiple host environments (hera, gaeac6, etc.)
+#   2. Two primary testing modes:
+#      - PR validation via standard test cases
+#      - CTest-based tests triggered via GitHub API
+#
+# Included files:
+#   - .gitlab-ci-ctests.yml: Handles CMake/CTest-based testing framework
+#   - .gitlab-ci-cases.yml: Defines templates for standard experiment cases
+#   - .gitlab-ci-hosts.yml: Contains host-specific configurations and test matrices
+#                          that can be extended to support additional hosts
+#
+# The host configuration in .gitlab-ci-hosts.yml is designed to be easily 
+# extended with new computing platforms and allows per-host specification 
+# of which test cases to run.
+
 stages:
   - build
   - create_experiments
   - run_tests
+  - finalize
 
-# Global variables
 variables:
-  BUILD: 'TODAY'
-  GIT_CLONE_PATH: '${CI_BUILDS_DIR}/${BUILD}/global-workflow'
+  BUILD: ''
+  CI_RUN_PATH: ${CI_BUILDS_DIR}/${BUILD}/${CI_COMMIT_SHORT_SHA}
+  GIT_CLONE_PATH: '${CI_RUN_PATH}/global-workflow'
   HOMEGFS: ${GIT_CLONE_PATH}
-  RUNTESTS_DIR: ${CI_BUILDS_DIR}/${BUILD}/RUNTESTS
+  RUNTESTS_DIR: ${CI_RUN_PATH}/RUNTESTS
   GIT_DEPTH: 1
   RUNNER_SCRIPT_TIMEOUT: 6h
   RUNNER_AFTER_SCRIPT_TIMEOUT: 6h
+  # Controls pipeline behavior: CTests (true) or PR cases (false)
+  GITHUB_API_TRIGGER: ${GITHUB_API_TRIGGER:-"false"}
+  PR_NUMBER: ${PR_NUMBER:-"0"}
+
+# Include specialized pipeline configuration files
+include:
+  - local: 'dev/ci/.gitlab-ci-ctests.yml'  # CTest framework configuration
+  - local: 'dev/ci/.gitlab-ci-cases.yml'   # Standard test case templates
+  - local: 'dev/ci/.gitlab-ci-hosts.yml'   # Host-specific configurations
```

Comment:
Since these are included explicitly, it is likely possible they do not need to be hidden.  May be we can put them in a directory `dev/ci/gitlab/includes/`?

Reply:
I don't think we should change the default one `.gitlab-ci.yaml` because that is the norm.  Let that be at the root of the project.  
If we change the default one (location and name), we will have to instruct the runner to find the correct one for this project.  That is extra hidden layer that the runner should not have to worry about.  It also means the same runner could potentially be used for other projects (if that is possible).

Reply:
The default location is at the root, so we have changed the location, but we do not have to.

Reply:
Insentiently, I think this is a good form factor that we should use as I intended to include a clean extendibility construct as soon as we go to three platforms.

Reply:
Ok I went with removing the hidden names for the imported pipelines but left the driver pipeline with the defaulted hidden name and left it in the re-assigned non-defaulted location under 'dev/ci' without changing the current settings in the GitLab controller.